spanner/spanner_snippets/spanner/integration_test.go

```diff
@@ -386,6 +388,23 @@ func runBackupSample(t *testing.T, f backupSampleFunc, dbName, backupID, errMsg
 	return b.String()
 }
 
+func runBackupSampleWithRetry(t *testing.T, f backupSampleFunc, dbName, backupID, errMsg string, maxAttempts int) string {
+	var b bytes.Buffer
+	success := testutil.Retry(t, maxAttempts, time.Minute, func(r *testutil.R) {
+		if err := f(&b, dbName, backupID); err != nil {
+			if spanner.ErrCode(err) == codes.InvalidArgument && strings.Contains(err.Error(), "Please retry the operation once the pending restores complete") {
+				r.Errorf("%s: %v", errMsg, err)
+			} else {
+				r.FailNowf("%s: %v", errMsg, err)
```

Comment:
Any particular reason we can't use `t.Fatalf` here? Do the retries matter at this point?

Reply:
I tested out that `t.Fatalf` does the right thing with the following test:
```go
func TestFoo(t *testing.T) {
	testutil.Retry(t, 5, 100*time.Millisecond, func(r *testutil.R) {
		t.Logf("Attempt: %v", r.Attempt)
		if r.Attempt == 4 {
			t.Fatalf("The 4th attempt failed!")
		}
		r.Errorf("Failure! %v", r.Attempt)
	})
}
```

The output is:
```
=== RUN   TestFoo
    license_test.go:95: Attempt: 1
    license_test.go:95: Attempt: 2
    license_test.go:95: Attempt: 3
    license_test.go:95: Attempt: 4
    license_test.go:97: The 4th attempt failed!
--- FAIL: TestFoo (0.30s)
```

Reply:
Argh. You're completely right, I don't know what I was thinking there. I've changed it into simply using `t.Fatalf`. Thanks for pointing it out.